src/components/entity/ha-chart-base.html

```diff
@@ -0,0 +1,228 @@
+<link rel='import' href='../../../bower_components/polymer/polymer-element.html'>
+
+<dom-module id="ha-chart-base">
+  <template>
+    <style>
+      canvas{
+        -moz-user-select: none;
+        -webkit-user-select: none;
+        -ms-user-select: none;
+      }
+      #chartTooltip {
+        opacity: 1;
+        position: absolute;
+        background: rgba(0, 0, 0, .7);
+        color: white;
+        border-radius: 3px;
+        -webkit-transition: all .1s ease;
+        transition: all .1s ease;
+        pointer-events: none;
+        -webkit-transform: translate(-50%, 0);
+        transform: translate(-50%, 0);
+      }
+      .chartjs-tooltip-key {
+        display: inline-block;
+        width: 10px;
+        height: 10px;
+        margin-right: 10px;
+      }
+    </style>
+    <div id="chartTarget" style="position: relative; height:200px; width:100%">
+      <canvas id="chartCanvas"></canvas>
+      <div id="chartTooltip">
+        <table></table>
+      </div>
+    </div>
+  </template>
+</dom-module>
+<script src="../../../bower_components/moment/moment.js"></script>
+<script src="../../../bower_components/chart.js/dist/Chart.js"></script>
+<script src="../../../bower_components/chartjs-chart-timeline/timeline.js"></script>
+<!--script src="../../../../chartjs-chart-timeline/timeline.js"></script-->
+<script>
+/* global Chart moment */// eslint-disable-line no-unused-vars
+class HaChartBase extends
+  Polymer.mixinBehaviors([Polymer.IronResizableBehavior], Polymer.Element) {
+  get chart() {
+    return this._chart;
+  }
+  static get is() { return 'ha-chart-base'; }
+  static get properties() {
+    return {
+      publish: {
+        type: Boolean,
+        observer: 'onPropsChange'
+      },
+      data: {
+        type: Object,
+        observer: 'onPropsChange'
+      },
+      selectedElement: {
+        type: Object,
+        readOnly: true,
+        notify: true,
+      },
+      connected: {
+        type: Object,
+        readOnly: false,
+        observer: 'onPropsChange'
```

Comment:
Make a single observer that will observe all the variables instead. Otherwise if both `data` and `connected` change together the observer would be called twice.